portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -70,4 +82,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   private String convertToJsonUsingGson(List<String> comments) {
     return new Gson().toJson(comments);
   }
+
+  private int getNumComments(HttpServletRequest request) { 
+    String numCommentsString = request.getParameter("numComments");
+
+    // Convert the input to an int.
+    int numComments;
+    try {
+      numComments = Integer.parseInt(numCommentsString);
```

Comment:
One thought here - you can reduce some of your boilerplate:

```
private int getNumComments(HttpServletRequest request) {
  String numCommentsString = request.getParameter("numComments");
  try {
    return numCommentsString == null || numCommentsString.isEmpty() ? Integer.MIN_VALUE : Integer.parseInt(numCommentsString);
  } catch (NumberFormatException e) {
    System.err.println(...);
    return Integer.MIN_VALUE;
  }
}
```

Your return statement is inlined and reduces code lines and variables.